zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -657,25 +656,27 @@ where
     // Store the transactions only after creating all of them. This avoids undesired
     // retransmissions in case a transaction is stored and the creation of a subsequent
     // transaction fails.
+    let mut transactions = Vec::with_capacity(proposal.steps().len());
     let mut txids = Vec::with_capacity(proposal.steps().len());
     #[allow(unused_variables)]
-    for (_, (build_result, outputs, fee_amount, utxos_spent)) in step_results {
+    for (_, (ref build_result, ref outputs, fee_amount, ref utxos_spent)) in step_results.iter() {
         let tx = build_result.transaction();
-        let sent_tx = SentTransaction {
+        transactions.push(SentTransaction {
             tx,
             created,
             account: account_id,
-            outputs,
-            fee_amount,
+            outputs: outputs.clone(),
```

Comment:
I don't know how to avoid this clone.

Reply:
`step_results.into_iter()~ should allow you to take everything by move.

Reply:
Tried that; the problem is that `build_result` doesn't live past the loop (and `tx` is a reference into the `BuildResult`).

It works fine with the clones, it's just inelegant.

Reply:
I fixed this. It required a small breaking change to `SentTransaction` but I think that's fine.